plugins/modules/hcloud_isos_info.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2022, Patrice Le Guyader
+# heavily inspired by the work of @LKaemmerling
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: hcloud_isos_info
+
+short_description: Gather infos about the Hetzner Cloud isos list.
+
+
+description:
+    - Gather infos about the Hetzner Cloud isos list.
+
+author:
+    - Patrice Le Guyader (@patlegu)
+    - Lukas Kaemmerling (@LKaemmerling)
+
+options:
+    id:
+        description:
+            - The ID of the iso image you want to get.
+        type: int
+    name:
+        description:
+            - The description of the iso you want to get.
+        type: str
+extends_documentation_fragment:
+- hetzner.hcloud.hcloud
+
+'''
+
+EXAMPLES = """
+- name: Gather hcloud isos type infos
+  hcloud_isos_info:
+  register: output
+
+- name: Print the gathered infos
+  debug:
+    var: output.hcloud_isos_info
+"""
+
+RETURN = """
+hcloud_isos_info:
+    description: The isos type infos as list
+    returned: always
+    type: complex
+    contains:
+        id:
+            description: ID of the ISO
+            returned: always
+            type: int
+            sample: 1937415
+        name:
+            description: Unique identifier of the ISO. Only set for public ISOs
+            returned: always
+            type: str
+            sample: fsn1
+        description:
+            description: DDescription of the ISO
+            returned: always
+            type: str
+            sample: Falkenstein DC Park 1
```

Comment:
While this is a string, and the description for ISOs are more or less free-text, I think this might confuse users. My suggestion uses a real description of a current ISO.

```suggestion
            sample: Debian 11.5 (amd64/netinstall)
```